### PR TITLE
Update tantivy and integrate the Columnar work

### PR DESCRIPTION
**Pull request by fulmicoton**
Update tantivy and integrate the Columnar work

- Fast field readers become `Column`. Column dynamically handle
  cardinality.
- Aggregation is getting an aggregation limits object.
- Minor changes.

Closes https://github.com/quickwit-oss/quickwit/issues/1548

### CODE REVIEW

File: quickwit/quickwit-search/src/filters.rs [hunk 1]
```diff
@@ -86,22 +87,27 @@ impl TimestampFilterBuilder {
         }
     }
 
+    /// None means that all
```
Comment (guilload) [inline quickwit/quickwit-search/src/filters.rs:90]: This comment is incomplete.

File: quickwit/quickwit-search/src/find_trace_ids_collector.rs [hunk 2]
```diff
@@ -231,25 +233,20 @@ impl SegmentCollector for FindTraceIdsSegmentCollector {
...
+                let span_timestamp = trace_id_term_ord.span_timestamp;
                 let found_term = self
-                    .inverted_index_reader
-                    .terms()
-                    .ord_to_term(trace_id.term_ord, &mut buffer)
-                    .expect("The term ord should exist in the term dict.");
-
+                    .trace_id_column
+                    .ord_to_str(trace_id_term_ord.term_ord, &mut buffer)
+                    .expect("Failed to lookup in the trace id column term dictionary");
```
Comment (guilload) [inline quickwit/quickwit-search/src/find_trace_ids_collector.rs:245]:
```suggestion
                    .expect("Failed to lookup trace ID in the column term dictionary");
```

File: quickwit/quickwit-search/src/leaf.rs [hunk 3]
```diff
@@ -226,85 +226,37 @@ async fn warm_up_postings(
...
+    futures::future::try_join_all(
+        columns
+            .into_iter()
+            .map(|col| async move { col.file_slice().read_bytes_async().await }),
+    )
+    .await?;
+    Ok(())
 }
 
+/// Populate the short-lived cache with the data for
```
Comment (guilload) [inline quickwit/quickwit-search/src/leaf.rs:245]:
```suggestion
/// Populates the short-lived cache with the data for
```

File: quickwit/quickwit-search/src/leaf.rs [hunk 4]
```diff
@@ -226,85 +226,37 @@ async fn warm_up_postings(
...
+/// all of the fast field passed in argument.
```
Comment (guilload) [inline quickwit/quickwit-search/src/leaf.rs:246]:
```suggestion
/// all of the fast fields passed as argument.
```

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs [hunk 5]
```diff
@@ -160,41 +161,28 @@ fn list_required_fields(field_mappings: &MappingTree) -> Vec<Field> {
...
-) -> anyhow::Result<()> {
-    if let Some(ref timestamp_field_name) = timestamp_field_name_opt {
-        let timestamp_field = schema
-            .get_field(timestamp_field_name)
-            .with_context(|| format!("Unknown timestamp field: `{timestamp_field_name}`"))?;
-
-        let timestamp_field_entry = schema.get_field_entry(timestamp_field);
-        if !timestamp_field_entry.is_fast() {
+fn validate_timestamp_field_if_any(builder: &DefaultDocMapperBuilder) -> anyhow::Result<()> {
+    let Some(timestamp_fieldname) = builder.timestamp_field.as_ref() else {
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit/quickwit-doc-mapper/src/default_doc_mapper/default_mapper.rs]: fixed

**Pull request merged**